### PR TITLE
fix: Proper setting of permissions for the tempdir on posix systems. Updated integ-test.

### PR DESCRIPTION
**Pull request by fl3x1324**
#462 

*Description of changes:Proper setting of permissions for the tempdir on posix systems. Updated integ-test.*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by fl3x1324** [comment]
@sanathkr Please give me some advice on the tests. Python is not really my thing. :) Sorry if code is messy.

**Comment by jfuss** [comment]
@fl3x1324 Can we just check in the jar instead of adding all the maven stuff?

**Comment by fl3x1324** [comment]
If you consider it a better move. I'm ok with it either way. Shall I create a new PR?

**Comment by jfuss** [comment]
@fl3x1324 Didn't realized you commented here. I pushed a commit to remove those pieces and it is building now. 

Previous to your commit, the integ tests were failing for me on mac as well (for java8).

File: samcli/local/lambdafn/runtime.py [hunk 1]
```diff
@@ -188,6 +188,9 @@ def _unzip_file(filepath):
 
     temp_dir = tempfile.mkdtemp()
 
+    if os.name == 'posix':
```
Comment (sanathkr) [inline samcli/local/lambdafn/runtime.py]: unit tests please :)

Comment (jfuss) [reply samcli/local/lambdafn/runtime.py]: Added unit tests

**Comment by sanathkr** [comment]
@fl3x1324 Thanks a ton for triaging the issue and submitting a PR! This was tricky one because it worked on some machines and didn't on others. And hey, congrats on your first PR 🎉

**Pull request merged**